File: core/src/ru/alexkulikov/firstfame/Player.java
package ru.alexkulikov.firstfame;


import com.badlogic.gdx.math.Vector2;
import com.badlogic.gdx.physics.box2d.BodyDef;
import com.badlogic.gdx.physics.box2d.PolygonShape;
import com.badlogic.gdx.physics.box2d.World;

public class Player extends GameObject {

    public Player(World world) {
        super(world);
    }

    public void createBody() {
        PolygonShape shape = new PolygonShape();
        shape.setAsBox(0.2f, 0.2f);
        Material material = Material.wood;
        createBody(shape, ObjectType.player, BodyDef.BodyType.DynamicBody, material.getRestitution(), material.getDensity(), material.getFriction());
    }

    public void jump(float power) {
        body.applyLinearImpulse(new Vector2(power * 0.8f, power * 2.5f), body.getPosition(), true);
    }
}

File: core/src/ru/alexkulikov/firstfame/objects/player/Player.kt
package ru.alexkulikov.firstfame.objects.player

import com.badlogic.gdx.math.Polygon
import com.badlogic.gdx.math.Vector2
import com.badlogic.gdx.physics.box2d.World
import ru.alexkulikov.firstfame.objects.GameObject

open class Player(world: World) : GameObject(world) {
    private val forceVector = Vector2(0f,0f)
    private val jumpVector = Vector2(0f,0f)

    open fun clearManual() {
        remove()
    }

    fun move(xPower: Float) = body.applyForceToCenter(forceVector.set(xPower, 0.0f), true)
    fun jumpSmall() = body.applyLinearImpulse(jumpVector.set(0.0f, 0.5f), body.position, true)

    val linearVelocity: Vector2
        get() = body.linearVelocity

    open fun overlaps(polygon: Polygon): Boolean = false
}